### DISCUSSION TITLE
v2.1.6

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#3118] [分享 / Share]
以下功能已不再维护 (不再修复 bug 和添加新功能), 但是你仍然可以为其提供 Pull Request.
- 直播间自动抽奖
- 默认播放器模式
- 播放器置顶
- 播放器定位

✨新增
- `清爽首页` 已追上 v1 的功能, 现正式开放使用:
  - 所有基础板块都已完成
  - 支持启用横向滚动
  - 支持设置最大宽度来改善 21:9 屏幕的体验
  - 热门与动态均支持在 `动态过滤器` 中设置的屏蔽关键词
  - 剩余未完成: 自定义板块布局, 极简首页
- 自定义顶栏:
  - 修改了搜索栏的样式, 和脚本整体设计风格更匹配.
  - 更新了创作中心相关的链接.
- `网址参数清理` 添加了 `-Arouter` 参数, 支持通过插件 (`urlParamsClean.tailingSlash`) 清理指定网址末尾的 `/`. (#2993, #3053)
- `下载视频` 面板支持滚动, 开始按钮固定在底部. (#2990)
- 排除了两个无用的 iframe:
  - *://message.bilibili.com/pages/nav/header_sync
  - *://s1.hdslb.com/bfs/seed/jinkela/short/cols/iframe.html

<details>
<summary>正式版将获得此前预览版中的新功能, 点击展开</summary>

- 支持 AV1 编码下载. (#2941)
- `自动更新器` 的选项中, 可以手动触发更新检查.
- 新增组件 `视频页默认定位` by [timongh](https://github.com/timongh). (PR #2960)
- `倍速记忆` 更名为 `倍速增强`, 功能分裂为 `启用倍速记忆` 和 `扩展倍速菜单`, 支持单独开启. (PR #2746)
> 这个只是过渡版本, PR #2746 还没完成, 不过应该快了吧(

- 对类似拜年祭的页面开启视频类功能支持. (#2999)
- 修复 `展开动态标题` 导致 `清爽首页` 中的视频卡片标题溢出. (#2944)
- 改进了 `动态反折叠` 的描述. (#2743)

</details>

🐛修复
- 修复了稍后再看中视频截图等扩展按钮出现两次.
- 修复 `启用视频截图` 在 WASM 播放器中静默失败, 现在将弹出提示. (PR #3047 by [ProjectXero](https://github.com/XeroAlpha))
- 修复 `隐藏首页横幅` 启用后页面高度不正确.
- 修复 `简化首页` 启用时对性能的意外损耗.
- 修复 `控制栏触摸优化` 失效.
- 修复 `简化直播间` 的头像大小调整失效. (#3084)
- 改进了动态首页中直播栏的高度计算. (#2211)
- 修复 `下载视频` 中更换输入源后清晰度列表没有更新. (#3069)
- 修复 `网址参数清理` 在 festival 类页面中复制出错误的链接. (#3060)
- 修复 `下载弹幕` 得到的 XML 弹幕没有转义. (#3055)
- 修复 `下载字幕` 失效. (#3073)
- 修复 `下载视频` 在合集类页面中失效. (#3045)
- 修复启用 `自定义顶栏` 后分区页面的横幅消失. (#3042)
- 修复收藏了纪录片后 `自定义顶栏` 的收藏弹窗异常. (#2973)
- `简洁至上` 合集包移除 `自动隐藏侧栏`, 因为实在太多人不看说明就装了.

☕开发者相关
- 修复 babel-loader 缓存导致组件的 commitHash 没更新.
- `ComponentOption.validator` 添加了第二个参数表示旧值, 并添加了 `getNumberValidator` API 方便快速生成一个校验数字的 `validator`.
- `DpiImage` 的 `size` 支持直接传入数字.
- 添加了两个 Sass Mixin: `absolute-h-center` `absolute-v-center`.
- 添加了 `enableHorizontalScroll` API.
- 添加了 `UpInfo` 组件, 可以用来展示 up 主信息.
- `isBwpVideo` 修改为异步函数, `BwpPlayerAgent` 合并至 `VideoPlayerAgent`, 由于检测 `bwp-video` 需要异步, 请避免在刚进入页面时调用 `playerAgent`. (#3046)

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.1.6'>v2.1.6</a>.</em>

**Comment by Geeyun-JY3** [comment]
想问问为啥这四个功能不维护了。
根据我的观察，感觉应该挺多人在用 `默认播放器模式`、`播放器定位` 和 `播放器置顶` 的。

**Reply by the1812** [reply]
没精力折腾自己从来不用的功能了, 除了 `直播间自动抽奖` 之后会删除 (因为早就没有奖可以抽了), 其他几个都是保留的, 谁有兴趣都可以维护下.

**Reply by Geeyun-JY3** [reply]
了解了。

**Comment by RX-02333** [comment]
真的期待极简首页，主要是直接有新番时间表可太方便了hh

**Reply by the1812** [reply]
时间表在清爽首页啊, 极简首页没有的

**Comment by the1812** [comment]
> 修复 下载视频 中更换输入源后清晰度列表没有更新.

这里引入了严格匹配编码的机制, 比如选择 HEVC 时如果只有 AVC 的源清晰度列表就没掉了, 感觉不太合适, 得再改改...

**Comment by Monokeros-zero** [comment]
清爽首页终于来了！番剧时间表展示时间太多，有点密集，没有之前清爽。

**Comment by FantasticAiming** [comment]
今天发生看一个诡异的事件  早上卸载了一些插件并且安装了一些新的  结果吃个饭回来后这一切都没发生过.....

**Comment by BranGao** [comment]
试了一下玩玩
iPadOS 15.4 配合userscript
能安装bilibili evolved 最新版
不过左侧没有任何东西😂

**Reply by Geeyun-JY3** [reply]
<https://github.com/the1812/Bilibili-Evolved/discussions/2717#discussioncomment-1839848>、#2926、#3106、
